osu.Game/Screens/Ranking/Statistics/PerformanceBreakdownChart.cs

```diff
@@ -0,0 +1,243 @@
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Origin = Anchor.TopCentre,
+                            Anchor = Anchor.TopCentre,
+                            ColumnDimensions = new[]
+                            {
+                                new Dimension(GridSizeMode.AutoSize),
+                                new Dimension(),
+                                new Dimension(GridSizeMode.AutoSize)
+                            }
+                        }
+                    }
+                }
+            };
+
+            spinner.Show();
+
+            new PerformanceBreakdownCalculator(playableBeatmap, difficultyCache, performanceCache)
+                .CalculateAsync(score, cancellationTokenSource.Token)
+                .ContinueWith(t => Schedule(() => setPerformanceValue(t.GetResultSafely())));
+        }
+
+        private void setPerformanceValue(PerformanceBreakdown breakdown)
+        {
+            spinner.Hide();
+            content.FadeIn(200);
+
+            var displayAttributes = breakdown.Performance.GetAttributesForDisplay();
+            var perfectDisplayAttributes = breakdown.PerfectPerformance.GetAttributesForDisplay();
+
+            setTotalValues(
+                displayAttributes.First(a => a.PropertyName == nameof(PerformanceAttributes.Total)),
+                perfectDisplayAttributes.First(a => a.PropertyName == nameof(PerformanceAttributes.Total))
+            );
+
+            var rowDimensions = new List<Dimension>();
+            var rows = new List<Drawable[]>();
+
+            foreach (PerformanceDisplayAttribute attr in displayAttributes)
+            {
+                if (attr.PropertyName == nameof(PerformanceAttributes.Total)) continue;
+
+                var row = createAttributeRow(attr, perfectDisplayAttributes.First(a => a.PropertyName == attr.PropertyName));
+
+                if (row != null)
+                {
+                    rows.Add(row);
+                    rowDimensions.Add(new Dimension(GridSizeMode.AutoSize));
+                }
+            }
+
+            chart.RowDimensions = rowDimensions.ToArray();
+            chart.Content = rows.ToArray();
+        }
+
+        private void setTotalValues(PerformanceDisplayAttribute attribute, PerformanceDisplayAttribute perfectAttribute)
+        {
+            achievedPerformance.Text = Math.Round(attribute.Value, MidpointRounding.AwayFromZero).ToLocalisableString();
+            maximumPerformance.Text = Math.Round(perfectAttribute.Value, MidpointRounding.AwayFromZero).ToLocalisableString();
+        }
+
+        [CanBeNull]
+        private Drawable[] createAttributeRow(PerformanceDisplayAttribute attribute, PerformanceDisplayAttribute perfectAttribute)
+        {
+            float percentage = (float)(attribute.Value / perfectAttribute.Value);
+            if (float.IsNaN(percentage))
```

Comment:
Usually we'd check the divisor ahead of time rather than post-checking for NaN.

Out of curiosity, what's the case where this can happen? A beatmap with no objects somehow reaching results?

Reply:
> Usually we'd check the divisor ahead of time rather than post-checking for NaN.

Alright, changed to a pre-check.

> Out of curiosity, what's the case where this can happen?

This is to hide attributes that have 0 as the maximum. e.g. if flashlight mod isn't on, flashlight bonus would always be 0.